server.go

```diff
@@ -82,6 +85,25 @@ func (s *BaseServer) worker(ctx context.Context, ch <-chan *Request, cb Callback
 	}
 }
 
+func (s *BaseServer) streamWorker(ctx context.Context, key ObjectID, stream Stream, cb EventCallback) {
+	fqn := fmt.Sprintf("%s.%s", key, stream.Name())
+	for event := range stream.Run(ctx) {
+		cb(fqn, event)
+	}
+}
+
+// StartStreams start the stream (events) workers in the background
+// use the ctx to cancel the streams workers
+func (s *BaseServer) StartStreams(ctx context.Context, cb EventCallback) {
+	s.m.RLock()
```

Comment:
Why the mutexes here ?

Reply:
Maps in go are not thread safe (for performance) so we need to make sure we have an exclusive access in case we want to "Read" or "Modify" the Map. otherwise we can get random panics. 

Even if the program flow will not cause any access race to the objects map probably it's better if we are safe.